pandas/core/algorithms.py

```diff
@@ -104,10 +104,11 @@ def _ensure_data(values, dtype=None):
 
             # ignore the fact that we are casting to float
             # which discards complex parts
-            with catch_warnings():
-                simplefilter("ignore", np.ComplexWarning)
-                values = ensure_float64(values)
-            return values, "float64", "float64"
+            # with catch_warnings():
+            #     simplefilter("ignore", np.ComplexWarning)
+            #     values = ensure_float64(values)
+            # return values, "float64", "float64"
+            return ensure_object(np.asarray(values)), "object", "object"
```

Comment:
we cant just return non-object values?  regardless, definitely dont leave the commented-out stuff above, just delete